File: musheng-tax-web/src/types/currency.ts
/**
 * 货币管理类型定义
 * 与后端 CurrencyController 接口对齐
 */

// 货币信息（对应后端 Currency 实体）
export interface Currency {
  id: number
  currencyCode: string      // 货币代码 USD/EUR/GBP/CAD
  currencyName: string      // 货币名称
  currencySymbol: string    // 货币符号 $, EUR, etc.
  decimalPlaces: number     // 小数位数
  status: number            // 状态 1-启用 0-禁用
  createTime: string
  updateTime?: string
}

// 货币查询参数（对应后端 CurrencyQueryRequest）
export interface CurrencyQuery {
  currencyCode?: string
  currencyName?: string
  status?: number
  page?: number
  size?: number
}

// 货币创建/更新参数（对应后端 CurrencyRequest）
export interface CurrencyForm {
  id?: number
  currencyCode: string
  currencyName: string
  currencySymbol: string
  decimalPlaces?: number
  status?: number
}

File: musheng-tax-system/musheng-config/src/main/java/com/musheng/config/currency/entity/Currency.java
package com.musheng.config.currency.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.musheng.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 货币实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_currency")
public class Currency extends BaseEntity {

    /**
     * 货币编码(USD/EUR/GBP/CAD)
     */
    private String currencyCode;

    /**
     * 货币名称
     */
    private String currencyName;

    /**
     * 货币符号($, €等)
     */
    private String currencySymbol;

    /**
     * 小数位数
     */
    private Integer decimalPlaces;

    /**
     * 状态(1启用, 0禁用)
     */
    private Integer status;

    /**
     * 货币对方向(DIRECT=XXX/CNY, REVERSE=CNY/XXX)
     * DIRECT: 外币对人民币（如 USD/CNY, EUR/CNY）
     * REVERSE: 人民币对外币（如 CNY/MOP, CNY/THB）
     */
    private String pairDirection;
}
